src/org/opensolaris/opengrok/web/api/v1/filter/LocalhostFilter.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * CDDL HEADER START
+ *
+ * The contents of this file are subject to the terms of the
+ * Common Development and Distribution License (the "License").
+ * You may not use this file except in compliance with the License.
+ *
+ * See LICENSE.txt included in this distribution for the specific
+ * language governing permissions and limitations under the License.
+ *
+ * When distributing Covered Code, include this CDDL HEADER in each
+ * file and include the License file at LICENSE.txt.
+ * If applicable, add the following below this CDDL HEADER, with the
+ * fields enclosed by brackets "[]" replaced with your own identifying
+ * information: Portions Copyright [yyyy] [name of copyright owner]
+ *
+ * CDDL HEADER END
+ */
+
+/*
+ * Copyright (c) 2018 Oracle and/or its affiliates. All rights reserved.
+ */
+package org.opensolaris.opengrok.web.api.v1.filter;
+
+import org.opensolaris.opengrok.logger.LoggerFactory;
+
+import javax.annotation.PostConstruct;
+import javax.servlet.http.HttpServletRequest;
+import javax.ws.rs.container.ContainerRequestContext;
+import javax.ws.rs.container.ContainerRequestFilter;
+import javax.ws.rs.container.PreMatching;
+import javax.ws.rs.core.Context;
+import javax.ws.rs.core.Response;
+import javax.ws.rs.ext.Provider;
+import java.io.IOException;
+import java.net.InetAddress;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+@Provider
+@PreMatching
+public class LocalhostFilter implements ContainerRequestFilter {
+
+    private static final Logger logger = LoggerFactory.getLogger(LocalhostFilter.class);
+
+    @Context
+    private HttpServletRequest request;
+
+    private final Set<String> localAddresses = new HashSet<>();
+
+    @PostConstruct
+    public void init() {
+        try {
+            localAddresses.add(InetAddress.getLocalHost().getHostAddress());
+            for (InetAddress inetAddress : InetAddress.getAllByName("localhost")) {
+                localAddresses.add(inetAddress.getHostAddress());
+            }
+        } catch (IOException e) {
+            logger.log(Level.SEVERE, "Could not get localhost addresses", e);
```

Comment:
this means the filter will not work. Is there any way how to bail out and cancel the deploy ?

Reply:
`System.exit(1)`? :D Or just API?

Reply:
I wonder what if the exception is left alone, possibly that would do the job ?

Reply:
If it throws an exception then:
```
28-Jun-2018 10:59:25.009 WARNING [RMI TCP Connection(2)-127.0.0.1] org.glassfish.jersey.internal.Errors.logErrors The following warnings have been detected: WARNING: Unknown HK2 failure detected:
MultiException stack 1 of 2
java.io.IOException: test
	at org.opensolaris.opengrok.web.api.v1.filter.LocalhostFilter.init(LocalhostFilter.java:55)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
...
```

And when I try to access the API then I get:
```
avax.servlet.ServletException: Servlet.init() for servlet [org.opensolaris.opengrok.web.api.v1.RestApp] threw exception
...
```

So `RestApp` won't work.

Reply:
Isn't it better to log the exception but not break the api and then check if the host list is not empty? so if this fails and it is empty 401 will be still returned

Reply:
This is how it works now, right? At first the set is empty so:
```java
if (!localAddresses.contains(request.getRemoteAddr())) {
            context.abortWith(Response.status(Response.Status.UNAUTHORIZED).build());
 }
```

will return `UNAUTHORIZED`

Reply:
Aha, should be fine then.

Reply:
:-D sorry. It's morning